packages/app-layout/AppLayout.tsx

```diff
@@ -0,0 +1,161 @@
+import { h, Component, prop } from 'skatejs';
+
+import {
+  GenericTypes,
+  matchMedia
+} from '../_helpers';
+
+import { HamburgerButton } from './components/HamburgerButton';
+import { Drawer } from './components/Drawer';
+import { Overlay } from './components/Overlay';
+import { Nav } from './components/Nav';
+import { NavContent } from './components/NavContent';
+
+import styles from './AppLayout.scss';
+
+const DEFAULT_RESPONSIBLE_WIDTH = '640px';
+
+type AppLayoutProps = Props & EventProps;
+
+type Attrs = {
+  'drawer-visible'?: boolean,
+  'responsive-width'?: string
+};
+
+type Props = {
+  drawerVisible?: boolean,
+  responsiveWidth?: string
+};
+
+// TODO: add onDrawerChange / onNarrowChange
+type EventProps = {};
+
+type Events = {};
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'bl-app-layout': GenericTypes.IntrinsicCustomElement<AppLayoutProps>
+      & GenericTypes.IntrinsicBoreElement<Attrs, Events>
+    }
+  }
+}
+
+export class AppLayout extends Component<AppLayoutProps> {
+
+  static get is() { return 'bl-app-layout'; }
+
+  static get props() {
+    return {
+      drawerVisible: prop.boolean( {
+        attribute: {
+          source: true
+        }
+      } ),
+      responsiveWidth: prop.string( {
+        initial: DEFAULT_RESPONSIBLE_WIDTH,
+        attribute: {
+          source: true
+        }
+      } ),
+      narrow: prop.boolean( {
+        attribute: {
+          source: true,
+        }
+      } )
+    };
+  }
+
+  drawerVisible: boolean;
+
+  narrow: boolean;
+
+  responsiveWidth: string;
+
+  connectedCallback() {
+
+    super.connectedCallback();
+
+    this.setupQueryMatchListener();
+
+  }
+
+  disconnectedCallback() {
+
+    super.disconnectedCallback();
+
+    this.matchMediaUnsubscribe();
```

Comment:
Ok, if it is recommended by skate...

I was going with the general feeling:
- let parent create/bootstrap what is needed, than enrich it
-- I might need intimated object
- when cleaning/destroying, clean what I did, than let parent clean his stuff
-- if parent's cleaning process would contain destroying the context somehow, to clean the mess, that might have been created
-- in other words, I might still need fully initiated object, when I am doing my cleanup...